### PR TITLE
Add the 3d deformation flow test case

### PR DESCRIPTION
**Pull request by szy21**
Co-authored-by: @Zhengyu-Huang @jiahe23 
Still needs
- [x] add plots
- [x] hook up CI
- [ ] run a high resolution simulation and make plots
- [x] squash and rebase

### CODE REVIEW

File: examples/3dsphere/deformation_flow.jl [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+    ∂ = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(0.0)),
+        top = Operators.SetValue(Geometry.WVector(0.0)),
+    )
+
+    hdiv = Operators.Divergence()
+
+    @. dydt.ρq1   = -∂(If(y.ρq1 .* w))
+    @. dydt.ρq1  -= hdiv(y.ρq1 .* uₕ)
```
Comment (szy21) [inline examples/3dsphere/deformation_flow.jl:169]: @simonbyrne Is this correct for scalar advection? uh and w are physical velocities and are on cell centers.

**Comment by szy21** [comment]
It runs but the result seems wrong. We would probably need some visualization (#354) for debugging.

**Review by jiahe23 — COMMENTED** [review]
mod() function modifies -180~0 to 180~360 and leave 0~180 unchanged. why was it replace by +180?

**Comment by szy21** [comment]
> mod() function modifies -180~0 to 180~360 and leave 0~180 unchanged. why was it replace by +180?

I was just trying to make it exactly the same as the test case in the paper (so longitude goes 0-360 rather than 180-360, 0-180). For a zonally symmetric simulation this probably doesn't matter, I'm not entirely sure though.

**Comment by szy21** [comment]
bors try

File: examples/3dsphere/deformation_flow.jl [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    dt = dt,
+    saveat = dt,
+    progress = true,
+    adaptive = false,
+    progress_message = (dt, u, p, t) -> t,
+)
+
+q1_error =
+    norm(sol.u[end].ρq1 ./ ρ.(coords.z) .- y0.ρq1 ./ ρ.(coords.z)) /
+    norm(sol.u[end].ρq1 ./ ρ.(coords.z))
```
Comment (jiahe23) [inline examples/3dsphere/deformation_flow.jl:214]: how about dividing the differences by the initial states?

Comment (jiahe23) [reply examples/3dsphere/deformation_flow.jl:214]: same for the other tracers

Comment (szy21) [reply examples/3dsphere/deformation_flow.jl]: done:)

**Review by jiahe23 — APPROVED** [review]

**Comment by szy21** [comment]
bors try

**Comment by valeriabarra** [comment]
Nice job! Should we add documentation of the formulation of this test case or provide a reference for where it is described?

**Comment by szy21** [comment]
> Nice job! Should we add documentation of the formulation of this test case or provide a reference for where it is described?

Good idea. Just added it.

**Comment by szy21** [comment]
bors try-

**Comment by szy21** [comment]
bors try

**Comment by szy21** [comment]
The artifacts look good. Thanks @jakebolewski for the plotting function!

**Pull request merged**